src/js/game/LevelMVC/LevelPlane.js

```diff
@@ -87,12 +87,51 @@ module.exports = class LevelPlane extends Array {
   * Changes the block at a desired position to the desired block.
   * Important note: This is the cornerstone of block placing/destroying.
   */
-  setBlockAt(position, block) {
+  setBlockAt(position, block, direction = null) {
     this[this.coordinatesToIndex(position)] = block;
+    let offset = [0,0];
+
+    // Direction will ever only not be null if we're calling this as a
+    // function of player movement.
+    switch (direction) {
+      case 0: {
+        offset[1] = -1;
+        break;
+      }
+      case 1: {
+        offset[0] = 1;
+        break;
+      }
+      case 2: {
+        offset[1] = 1;
+        break;
+      }
+      case 3: {
+        offset[0] = -1;
+        break;
+      }
+    }
+
+    let positionInQuestion = [0,0];
+    // This will either be the pos the player is leaving or entering, depending on situation
+    if (this.levelModel) {
+      positionInQuestion = [this.levelModel.player.position[0] + offset[0], this.levelModel.player.position[1] + offset[1]];
+    }
+    let wasOnADoor = false;
+    // If the questionable position was a door, we want to do a few things differently.
+    if (this[this.coordinatesToIndex(positionInQuestion)].blockType === "doorIron") {
```

Comment:
Can this be `this.getBlockAt(positionInQuestion)`?  Let's avoid working in indices as much as possible.

Reply:
Ha, absolutely. While the gears were turning, I forgot we had a helper for that. 😂